src/profiler/profiler.c

```diff
@@ -411,29 +432,32 @@ void xdebug_profiler_function_end(function_stack_entry *fse)
 	/* use previously created filename and funcname (or a reference to them) to show
 	 * time spend */
 	if (fse->user_defined == XDEBUG_BUILT_IN) {
-		char *tmp_key = xdebug_sprintf("php::%s", fse->profiler.funcname);
-		char *fl_ref = NULL, *fn_ref = NULL;
+		size_t tmp_key_funcname_len = strlen(fse->profiler.funcname);
 
-		fl_ref = get_filename_ref((char*) "php:internal");
-		fn_ref = get_functionname_ref(tmp_key);
+		memcpy(tmp_key + TMP_KEY_PREFIX_LEN,
+			fse->profiler.funcname,
+			tmp_key_funcname_len > TMP_KEY_MAX_LEN ? TMP_KEY_MAX_LEN : tmp_key_funcname_len + 1
```

Comment:
this should be moved to macro and we should suffix with `...` when over buffer limit

Reply:
I disagree with suffixing with `...` as that would mean 3 fewer characters as unique index. The UI aspects in Kcachegrind are also not going to cope with *that* much data either to display.

I also think (hope!) that 1024 for a fully qualified function name is likely to large anyway. 1024 characters is quite a lot for a function name, not even Symfony goes that far!